src/main/kotlin/com/hibob/academy/dao/OwnerDao.kt

```diff
@@ -21,46 +20,73 @@ class OwnerTable(tableName: String = "owner") : JooqTable(tableName) {
     }
 }
 
+@Component
+class OwnerMapper : RecordMapper<Record, Owner> {
+    override fun map(record: Record): Owner {
+        val nameParts = record.getValue("name", String::class.java)?.split("\\s+".toRegex())
+        val firstName = nameParts?.getOrNull(0).orEmpty()
+        val lastName = nameParts?.drop(1)?.joinToString(" ").takeIf { it?.isNotEmpty() == true }
+
+        return Owner(
+            id = record.getValue("id", Long::class.java),
+            name = record.getValue("name", String::class.java),
+            companyId = record.getValue("company_id", Long::class.java),
+            employeeId = record.getValue("employee_id", String::class.java),
+            firstName = firstName,
+            lastName = lastName
+        )
+    }
+}
+
 @Component
 class OwnerDao(
     private val sql: DSLContext
 ) {
 
     private val ownerTable = OwnerTable.instance
 
-    private val ownerMapper = RecordMapper<Record, Owner>
-    { record ->
-
-        val nameParts = record[ownerTable.name]?.split("\\s+".toRegex())
-        val firstName = nameParts?.getOrNull(0).orEmpty()
-        val lastName = nameParts?.drop(1)?.joinToString(" ").takeIf { it?.isNotEmpty() == true }
+    companion object {
+        val ownerMapper = RecordMapper<Record, Owner>
+        { record ->
+            val nameParts = record.getValue("name", String::class.java)?.split("\\s+".toRegex())
+            val firstName = nameParts?.getOrNull(0).orEmpty()
+            val lastName = nameParts?.drop(1)?.joinToString(" ").takeIf { it?.isNotEmpty() == true }
 
-        Owner(
-            id = record[ownerTable.id],
-            name = record[ownerTable.name],
-            companyId = record[ownerTable.companyId],
-            employeeId = record[ownerTable.employeeId].toString(),
-            firstName = firstName,
-            lastName = lastName
-        )
+            Owner(
+                id = record.getValue("id", Long::class.java),
+                name = record.getValue("name", String::class.java),
+                companyId = record.getValue("company_id", Long::class.java),
+                employeeId = record.getValue("employee_id", String::class.java),
+                firstName = firstName,
+                lastName = lastName
+            )
+        }
```

Comment:
this whole sections is a duplicated code - why not user the mapper itself? It seems identical - or am I missing something?

Reply:
No, it was my mistake... I forgot to delete the ownerMapper class